index.js

```diff
@@ -17,9 +17,16 @@ function handler(event) {
   }
 
   // just kick out invalid looking paths
-  // either: /podcast_id/episode_guid/digest/filename.mp3
-  //     or: /podcast_id/feed_id/episode_guid/digest/filename.mp3
-  if (parts.length !== 4 && parts.length !== 5) {
+
+  // /podcast_id/episode_guid/digest/filename.mp3
+  // /podcast_id/feed_id/episode_guid/digest/filename.mp3
+  // /t/a/path/<file>/<format>.<extension>
+  // /t/<host>/[<path>/]<file>/<format>.<extension>
```

Comment:
Would it be cleaner to split everything after removing the optional `/usw2/` / `/use1/` prefix... into separate `handleDigest` / `handleTranscode` methods?

It seems like the validations between the 2 are fairly distinct... branching on `parts[0] === 't'`.

(It does seem like you were trying to keep the code flow the same, wrapping stuff individually in `if`s ... but I think the git diff will end up pretty big either way).